api/src/wfl/executor.clj

```diff
@@ -473,74 +475,61 @@
          (remove nil?))))
 
-(defn ^:private terra-executor-workflows-by-filters
+(defn ^:private terra-executor-workflows
   "Return all the non-retried workflows executed by `executor`
   matching specified `filters`."
   [tx {:keys [details] :as executor} filters]
   (postgres/throw-unless-table-exists tx details)
-  (->> (terra-executor-workflows-by-filters-sql-params executor filters)
-       (jdbc/query tx)
-       (terra-workflows-from-records executor)))
-
+  (let [sql-params      (terra-executor-workflows-sql-params executor filters)
+        executor-subset (select-keys executor [:workspace :methodConfiguration])]
+    (map #(merge executor-subset %) (jdbc/query tx sql-params))))
+
+;; 1. Presently Rawls doesn't support submitting a subset of a snapshot:
+;;    If we wish to retry 1+ workflows stemming from a snapshot,
+;;    we must resubmit the snapshot in its entirety.
+;; 2. Though we are technically resubmitting snapshot references
+;;    and not submissions, this query takes a submission-based approach.
+;;    Why? Because there is no ambiguity when linking a submission
+;;    to its workflows.  With the capacity for retries, one reference
+;;    may map to multiple sets of workflows, and retrying the reference
+;;    should only update the sibling workflow records of those supplied.
+;;
 (defn ^:private workflow-and-sibling-records
   "Return the workflow records for all workflows in submissions
   associated with the specified `workflows` --
   i.e. records for `workflows` and their siblings."
   [tx {:keys [details] :as executor} workflows]
   (postgres/throw-unless-table-exists tx details)
-  (let [workflow-ids (util/to-quoted-comma-separated-list (map :uuid workflows))
-        _            (-> (str "%s For workflows %s, "
-                              "fetching sibling workflow records by submission")
-                         (format (log-prefix executor) workflow-ids)
-                         log/debug)
-        ;; 1. Presently Rawls doesn't support submitting a subset of a snapshot:
-        ;;    If we wish to retry 1+ workflows stemming from a snapshot,
-        ;;    we must resubmit the snapshot in its entirety.
-        ;; 2. Though we are technically resubmitting snapshot references
-        ;;    and not submissions, this query takes a submission-based approach.
-        ;;    Why? Because there is no ambiguity when linking a submission
-        ;;    to its workflows.  With the capacity for retries, one reference
-        ;;    may map to multiple sets of workflows, and retrying the reference
-        ;;    should only update the sibling workflow records of those supplied.
-        query "SELECT * FROM %s
-               WHERE submission IN
-                 (SELECT DISTINCT submission FROM %s
-                 WHERE workflow IN %s)"]
+  (let [workflow-ids (util/to-quoted-comma-separated-list
+                      (map :workflow workflows))
+        query        (str/join \space ["SELECT * FROM %s"
```

Comment:
Here's how this change manifests itself in the logs (none of the large `\n        ` gaps in between lines as before):

```
{"severity":"DEBUG","logging.googleapis.com/sourceLocation":{"file":"wfl/executor.clj","line":533},"timestamp":"2022-01-11T19:50:36.720255Z","message":"jdbc/query: okotsopo@jdbc:postgresql:wfl#2287 \"SELECT * FROM TerraExecutor_000000664 WHERE submission IN (SELECT DISTINCT submission FROM TerraExecutor_000000664 WHERE workflow IN ('99956c17-26af-4d50-8954-9de9ecc4f733','1286c5b1-6ad9-4881-a033-c32f26a84550'))\""}
```

Reply:
Thank you for doing this. I've wanted it for a while.